src/compiler/compiler.ml

```diff
@@ -221,17 +221,17 @@ module Setup = struct
 				else
 					Printf.sprintf "(%s) %s" wobj.w_name msg
 				in
-				message ctx (msg,p,DKCompilerMessage,Warning)
+				message ctx (msg,p,depth,DKCompilerMessage,Warning)
 			| WMDisable ->
 				()
 		);
 		com.error <- error ctx;
-		let filter_messages = (fun keep_errors predicate -> (List.filter (fun ((_,_,_,sev) as cm) ->
+		let filter_messages = (fun keep_errors predicate -> (List.filter (fun ((_,_,_,_,sev) as cm) ->
```

Comment:
As a future TODO, I really prefer records over these terror tuples. My personal cutoff is more than three items.